cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1407,6 +1410,76 @@ func maybeMountSave(disk disks.Disk, rootdir string, encrypted bool, mountOpts *
 	return true, nil
 }
 
+func generateMountsModeClassic(mst *initramfsMountsState, label string) error {
+	// Mount ESP
+	// Must be where the kernel came from, without fallbacks
+	espuuid, err := bootFindPartitionUUIDForBootedKernelDisk()
+	if err != nil {
+		return err
+	}
+	espSrc := filepath.Join("/dev/disk/by-partuuid", espuuid)
+	espOpts := &systemdMountOptions{
+		// vfat needs fsck to mount ESP that was unsafely
+		// unmounted
+		NeedsFsck: true,
+		// do not persist mount into running system, otherwise
+		// regular fstab cannot mount ESP at /boot/efi as
+		// expected
+		Ephemeral: true,
+	}
+	if err := doSystemdMount(espSrc, boot.InitramfsUbuntuSeedDir, espOpts); err != nil {
+		return err
+	}
+
+	// get the disk that we mounted the ESP from as a reference
+	// point for future mounts
+	disk, err := disks.DiskFromMountPoint(boot.InitramfsUbuntuSeedDir, nil)
+	if err != nil {
+		return err
+	}
+
+	// TODO measure model, if we will have one
```

Comment:
this TODO might not be relevant if this function doesn't try to stay generic for the time being. At this point is probably premature for that anyway.

Reply:
I'll drop it then.